lib/strong_migrations/checker.rb

```diff
@@ -170,6 +173,13 @@ def change
         end
       end
 
+      remove_index_index = @ordered_migrations.index { |k| k.first == :remove_index }
+      add_index_index = @ordered_migrations.drop(remove_index_index).index { |k| k.first == :add_index } if remove_index_index.present?
+
+      if remove_index_index.present? && add_index_index
+        raise_error :dangerous_replace_index
+      end
```

Comment:
I wasn't so sure where to put this check, so I just added it here for discussion. It would be nice if there was a class for each "checker" that could raise error and have its own methods for this sort of stuff.

Also, currently it doesn't check if you're removing and adding the index for the same table, do you think that's something that should be added?

Do you have any suggestion here?